### PR TITLE
use restore feature to refine codes releated ddl

### PR DESCRIPTION
**Pull request by GregoryIan**
### What problem does this PR solve? 
use restore feature to refine codes releated ddl

### What is changed and how it works?
use restore feature to implement rename and split ddl

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by csuzhangxc** [comment]
@GregoryIan resolve the conflict.

**Comment by GregoryIan** [comment]
I would do it after #50 merged

**Comment by GregoryIan** [comment]
@csuzhangxc @amyangfei PTAL

File: syncer/ddl_test.go [hunk 1]
```diff
@@ -373,22 +349,25 @@ func (s *testSyncerSuite) TestResolveGeneratedColumnSQL(c *C) {
...
+		sqls, _, err := syncer.resolveDDLSQL(parser, ast1, "test")
 		c.Assert(err, IsNil)
 
 		c.Assert(len(sqls), Equals, 1)
 		getSQL := sqls[0]
 		c.Assert(getSQL, Equals, tc.expected)
 
-		ast2, err := parser.ParseOneStmt(getSQL, "", "")
-		c.Assert(err, IsNil)
+		// FIXME: remove me after implement generated function in restore
```
Comment (amyangfei) [inline syncer/ddl_test.go]: latest parser has fixed this bug, we can remove these comments now

File: test.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
+package main
```
Comment (amyangfei) [inline test.go]: is this file used for testing only, can remove now?

Comment (GregoryIan) [reply test.go]: yep

File: syncer/ddl_test.go (hunk 1 above)
Comment (GregoryIan) [reply syncer/ddl_test.go]: it's useless, because of some very tricky code,  I decide remove these code

File: syncer/filter.go [hunk 3]
```diff
@@ -128,7 +110,7 @@ func (s *Syncer) skipQuery(tables []*filter.Table, stmt ast.StmtNode, sql string
 
 	if len(tables) > 0 {
 		tbs := s.bwList.ApplyOn(tables)
-		if len(tbs) == 0 {
+		if len(tbs) != len(tables) {
```
Comment (amyangfei) [inline syncer/filter.go]: suppose we rename a table from name `old` to name `new`, and table `old` is in `do-tables` and `new` is new `ignore-tables`.
should we filter this rename DDL?

Comment (GregoryIan) [reply syncer/filter.go]: yep

**Comment by GregoryIan** [comment]
@csuzhangxc @july2993 PTAL

File: pkg/parser/common.go [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+	return bf.String(), nil
+}
+
+// SplitDDL splits multiple operations in one DDL statement into multiple DDL statements
+// if fail to restore, it would not restore the value of `stmt` (it changes it's values if `stmt` is one of  DropTableStmt, RenameTableStmt, AlterTableStmt)
+func SplitDDL(stmt ast.StmtNode, schema string) (sqls []string, err error) {
+	var (
+		b          []byte
+		schemaName = model.NewCIStr(schema) // fill schema name
+		bf         = bytes.NewBuffer(b)
```
Comment (csuzhangxc) [inline pkg/parser/common.go]: `bf         buffer.Buffer` or `bf         = new(bytes.Buffer)` is clearer?

File: pkg/parser/common.go [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+		for _, t := range tables {
+			if t.Schema.O == "" {
+				t.Schema = schemaName
+			}
+
+			v.Tables = []*ast.TableName{t}
+			bf.Reset()
+			err = stmt.Restore(ctx)
+			if err != nil {
+				return nil, errors.Annotate(err, "restore ast node")
```
Comment (csuzhangxc) [inline pkg/parser/common.go]: is it needed to `v.Tables = tables` before this `return`? similar work for other cases.

File: pkg/parser/common.go [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+			}
+			sqls = append(sqls, bf.String())
+
+			if spec.Tp == ast.AlterTableRenameTable {
+				v.Table = spec.NewTable
+			}
+		}
+		v.Specs = specs
```
Comment (csuzhangxc) [inline pkg/parser/common.go]: whether needing to reset `v.Table` here?

File: pkg/parser/common.go [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+
+		return sqls, nil
+
+	default:
+		return nil, errors.Errorf("unkown type ddl %+v", stmt)
```
Comment (csuzhangxc) [inline pkg/parser/common.go]: s/unkown/unknown

File: pkg/parser/common.go (hunk 5 above)
Comment (GregoryIan) [reply pkg/parser/common.go]: according to comment in L179: if fail to restore, it would not restore the value of `stmt`.  but we can do it, let me see

**Pull request merged**